### DISCUSSION TITLE
Need help configuring SMTP and ADMIN_TOKEN

### DISCUSSION THREAD

**Discussion by Aryaj07** [dani-garcia/vaultwarden#5266] [Q&A]
`version: '3'

services:
  vaultwarden:
    restart: always
    container_name: vaultwarden
    image: vaultwarden/server:latest
    volumes:
      - /home/vaultwarden/data/:/data/
    ports:
      - 8080:80
    environment:
      ADMIN_TOKEN: "token with $$ inside these quotes"
      SMTP_HOST: mail.google.com
      SMTP_FROM: mail
      SMTP_FROM_NAME: VaultWarden
      SMTP_SECURITY: starttls
      SMTP_PORT: 587
      SMTP_USERNAME: mail
      SMTP_PASSWORD: pwd
      SMTP_TIMEOUT: 30
      SMTP_AUTH_MECHANISM: "Plain"
      LOGIN_RATELIMIT_MAX_BURST: 10
      LOGIN_RATELIMIT_SECONDS: 60
      DOMAIN: https://domain.org
      INVITATION_ORG_NAME: HomeVault
      INVITATIONS_ALLOWED: "true"
      SIGNUPS_ALLOWED: "true"
      SIGNUPS_DOMAINS_WHITELIST: google.com
      SIGNUPS_VERIFY: "true"
      SIGNUPS_VERIFY_RESEND_TIME: 3600
      SIGNUPS_VERIFY_RESEND_LIMIT: 6
      EMERGENCY_ACCESS_ALLOWED: "true"
      SENDS_ALLOWED: "true"
      WEB_VAULT_ENABLED: "true"`
      
      
      
      
So this is my compose file and when i try to run this in docker first issue it gives me is i cannot register an account 2nd i cannot access the admin panel i use `docker inspect vaultwarden` to get the ADMIN_TOKEN from the config so i would like to know what i m doing wrong....

**Comment by stefan0xC** [comment]
> i cannot register an account

What happens? What do the logs say? Is there an error message?

> i cannot access the admin panel

Are you entering the password you have used when generating the `ADMIN_TOKEN`? Also what method did you use to generate it?

> `ADMIN_TOKEN: "token with $$ inside these quotes"`
> ...
>  i use `docker inspect vaultwarden` to get the ADMIN_TOKEN from the config

Are you entering this `ADMIN_TOKEN` as the password? Because that won't work if that is [the argon2id PHC string](https://github.com/dani-garcia/vaultwarden/wiki/Enabling-admin-page#secure-the-admin_token).

**Reply by Aryaj07** [reply]
> What happens? What do the logs say? Is there an error message?

That was a mistake on my end i kept google.com as allowed domain to register but thats gone and i can create a user now no issues there


> Are you entering the password you have used when generating the `ADMIN_TOKEN`? Also what method did you use to generate it?

this is the argon string i setup in the docker compose file :-

`ADMIN_TOKEN: $$argon2id$$v=19$$m=19456,t=2,p=1$$YkxMWW5MdlhBMGsvVG1TUGNqbW5ramgrU0w4Y1ZGNGFKcUZqcVhUMFdvQT0$$veHcv+MDAHEqodgCsCAu7XDeggVk6BAYSkiIqq3hya8`

and this is the token i get in the when i do `docker inspect vaultwarden` ----> `"ADMIN_TOKEN=$argon2id$v=19$m=19456,t=2,p=1$YkxMWW5MdlhBMGsvVG1TUGNqbW5ramgrU0w4Y1ZGNGFKcUZqcVhUMFdvQT0$veHcv+MDAHEqodgCsCAu7XDeggVk6BAYSkiIqq3hya8"`

whatever i copy after the = should be the admin token but that doesnt work

and i cannot seem to send mail 


`[2024-12-10 07:31:24.062][request][INFO] POST /identity/accounts/register`
`[2024-12-10 07:31:54.463][vaultwarden::mail][ERROR] SMTP error: Connection error: Network is unreachable (os error 101)`
`[2024-12-10 07:31:54.465][vaultwarden::api::core::accounts][ERROR] Error sending welcome email: SMTP error: Connection error: Network is unreachable (os error 101)`
`[2024-12-10 07:31:54.522][response][INFO] (identity_register) POST /identity/accounts/register => 200 OK`

these are the logs when i created a user since signup verify is true it says verify the email before login.

i think that sums it up

**Reply by stefan0xC** [reply]
> whatever i copy after the = should be the admin token but that doesnt work

Nope, that's not how password hashing works. You want to enter the password you used to generate the hash not the hash itself.

**Reply by Aryaj07** [reply]
> > whatever i copy after the = should be the admin token but that doesnt work
> 
> 
> 
> Nope, that's not how password hashing works. You want to enter the password you used to generate the hash not the hash itself.



then what should the password be? i m confuse

**Reply by BlackDex** [reply]
How did you generate the hash/token?

**Reply by Aryaj07** [reply]
> How did you generate the hash/token?

basically i used argon2 cli and 

"Mysecretpassword" was changed to a password i wanted from the github documentation

**Reply by BlackDex** [reply]
Then use that password!

**Reply by Aryaj07** [reply]
> Then use that password!

omg i m dumb i thought we need to enter the argon string T_T



well since that is sorted out the other problem i have is smtp 

[2024-12-10 09:01:17.877][request][INFO] POST /admin/test/smtp
[2024-12-10 09:01:19.069][request][INFO] POST /admin/test/smtp
[2024-12-10 09:01:47.938][vaultwarden::mail][ERROR] SMTP error: Connection error: Network is unreachable (os error 101)
[2024-12-10 09:01:47.940][response][INFO] (test_smtp) POST /admin/test/smtp => 400 Bad Request
[2024-12-10 09:01:49.147][vaultwarden::mail][ERROR] SMTP error: Connection error: Network is unreachable (os error 101)
[2024-12-10 09:01:49.149][response][INFO] (test_smtp) POST /admin/test/smtp => 400 Bad Request

what might this error be? i m using portainer and docker compose.

**Reply by Aryaj07** [reply]
Okay nvm i set the wrong smtp host so that is also sorted i'll just close this discussion
